magic_marker/magic_marker.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+
+"""MagicMarker is a tool that automatically marks pytests with a UUID"""
+# ======================================================================================================================
+# Imports
+# ======================================================================================================================
+from __future__ import absolute_import
+from flake8.main.cli import main as flake8_main
+from contextlib import contextmanager
+import sys
+import six
+import json
+import uuid
+import re
+import tempfile
+import string
+import random
+import shutil
+import errno
+import os
+
+
+class MagicMarker(object):
+
+    def __init__(self, mark):
+        """Crate a new MagicMarker object
+
+        Args:
+            mark (str): The mark to create
+        """
+        self._mark = mark
+        dir_name = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(10))
+        self._backup_dir = os.path.join(tempfile.gettempdir(), dir_name)
```

Comment:
You should just use [mkdtemp](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp) instead.